src/rust/catnap/mod.rs

```diff
@@ -219,9 +219,21 @@ impl CatnapLibOS {
     }
 
     /// Sets a socket as a passive one.
-    pub fn listen(&mut self, qd: QDesc, backlog: usize) -> Result<(), Fail> {
+    pub fn listen(&mut self, qd: QDesc, mut backlog: usize) -> Result<(), Fail> {
         trace!("listen() qd={:?}, backlog={:?}", qd, backlog);
 
+        // Check if backlog length is valid.
+        if backlog > libc::SOMAXCONN as usize {
```

Comment:
Well, I advocate us to interpose, specially because this is implementation defined and:
- Demikernel has a significant lower limit than Linux for pending connections by construction. For instance, in Catloop this depends on the number of memory queues that may be created system-wide. There are also constraints on this for Catcollar and Catnip. Eventually we will have our own limit.

Reply:
Yes, it seems like according to the POSIX standard, we should silently truncate the value to some max that we define, instead of returning an error. We should also do a debug log.

Reply:
Sure, I'll do.